### PR TITLE
Add/update HWRF physics

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR adds missing HWRF physics parameterizations and updates the Ferrier-Aligo microphysics:
- WRFv4 Noah LSM
- GFDL surface layer
- Unified HEDMF PBL
- icloud=4 option in RRTMG

In addition, this PR:
- cleans up old/unused/unnecessary HWRF regression tests
- adds the missing `effr_in=.true.` for all Thompson MP based runs (there is now a guard in fv3atm's `GFS_typedefs.F90` to prevent running Thompson MP with `effr_in=.false.`
- shortens the runtime for the `fv3_ccpp_rrfs_v1beta` run from 48h to 24h
- in ccpp-physics: update `tsfc` correctly when there is ice on open water grid points (fixes https://github.com/NCAR/ccpp-physics/issues/515)
- set `OMP_STACKSIZE` in orion and wcoss_dell_p3 job submission scripts to avoid errors with threaded tests

### How were these changes tested?

Regression tests where run prior to pulling in #217 against the existing EMC baselines. All regression tests that were expected to pass passed, all tests that were expected to fail (because of changes to the physics or for newly added tests) failed (but ran to completion).

I also reverted the change of cloud effective radii in Thompson MP and ran the tests against the existing baseline. After reverting this cloud effective radii update, the regression tests using Thompson MP also passed against the existing baseline. The cloud effective radii update has no real effect on the results (except b4b differences) and is required for the `icloud=4` option to run.

[rt_hera_intel.log](https://github.com/ufs-community/ufs-weather-model/files/5423671/rt_hera_intel.log)
[rt_hera_intel_fail_test.log](https://github.com/ufs-community/ufs-weather-model/files/5423672/rt_hera_intel_fail_test.log)
[rt_hera_gnu.log](https://github.com/ufs-community/ufs-weather-model/files/5423673/rt_hera_gnu.log)
[rt_hera_gnu_fail_test.log](https://github.com/ufs-community/ufs-weather-model/files/5423674/rt_hera_gnu_fail_test.log)
[rt_hera_intel_without_module_mp_thompson_change.log](https://github.com/ufs-community/ufs-weather-model/files/5423677/rt_hera_intel_without_module_mp_thompson_change.log)
[rt_hera_intel_without_module_mp_thompson_change_fail_test.log](https://github.com/ufs-community/ufs-weather-model/files/5423679/rt_hera_intel_without_module_mp_thompson_change_fail_test.log)

I ran the tests again after pulling in #217 and #219 against the official baseline. The same tests failed for hera.intel than before - OK. Update: I can now run the hera.gnu tests as well, after having identified the problem (see https://github.com/ufs-community/ufs-weather-model/issues/235).

### Are the changes covered by regression tests?

New regression tests have been added for the HWRF suite for a global C96 grid and a regional 25km ESG grid. Note that the HWRF suite currently runs with Thompson MP, since Ferrier-Aligo crashes in both global and regional runs ever since the FV3 dycore and FMS were updated several months ago (the current version in ccpp-physics is also broken). @yangfanglin and @ericaligo-NOAA know about this problem and we are working on a solution.

### Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

Full regression tests will be run again and new baselines will be created on all tier-1 platforms when this PR is supposed to go in. Tests will also be run on cheyenne.intel and cheyenne.gnu, unless support for these systems has been broken in the recent ufs-weather-model/ufs-s2s-model merge efforts (these have not been regression tested on cheyenne).

**Update 2020/11/05**

The addition of the tsfc one-liner change in ccpp-physics changes the regression test results for all tests. New baselines were created and regression tests passed against these baselines on the following systems:
- hera.gnu
- hera.intel
- wcoss_cray
- orion.intel
- wcoss_dell_p3
- cheyenne.intel
- cheyenne.gnu

## Dependencies

https://github.com/ufs-community/ufs-weather-model/pull/223
https://github.com/NOAA-EMC/fv3atm/pull/188
https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/43
https://github.com/NCAR/ccpp-physics/pull/509

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
Looks good.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by BinLiu-NOAA — APPROVED** [review]
Looks good to me.

File: tests/fv3_conf/ccpp_c96_HAFS_v0_hwrf_run.IN [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+cp    @[RTPWD]/FV3_input_data/*grb .
+cp    @[RTPWD]/FV3_input_data/*_table .
+cp    @[RTPWD]/FV3_input_data/*configure .
+
+# Copy landuse/soil/vegetation parameter tables for HWRF Noah LSM
+cp    @[RTPWD]/FV3_input_data_hafs/GENPARM.TBL .
+cp    @[RTPWD]/FV3_input_data_hafs/SOILPARM.TBL .
+cp    @[RTPWD]/FV3_input_data_hafs/VEGPARM.TBL .
+
+# Copy diag table, depending on microphysics choice
```
Comment (junwang-noaa) [inline tests/fv3_conf/ccpp_c96_HAFS_v0_hwrf_run.IN:39]: so HWRF are running with different MPs?

Comment (climbfuji) [reply tests/fv3_conf/ccpp_c96_HAFS_v0_hwrf_run.IN:39]: It runs with Thompson MP and FA-MP. It may also run with GFDL MP, but that hasn't been tested.

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**